### ISSUE TITLE
Grammar Checker?

### ISSUE THREAD

**Issue by Mehul** [RickStrahl/MarkdownMonster#378]
Hi Rick,

Do you have any grammar checking capabilities (now or future planned) in MDM? 

If not, could you use something like write good? https://github.com/btford/write-good

I saw it used here: https://github.com/twhite96/checkyoself though it'd be great if it was part of MDM. Please? :)

**Comment by RickStrahl** [comment]
I've looked at this stuff before, but it can't be an online service as that would just be too slow to keep up with the editor as you type. So it would have to be something local and I haven't seen anything that I can plug into a local .NET app for this.

Also due to the editor's architecture it's going to be very difficult to correlate multiple words to grammar checked text, due to the embedded formatting that the grammar checker likely won't understand. Spelling is tedious but relatively easy to implement because it works on single words so MM can basically check one line at a time. But multi-line means deparsing the markdown, checking grammar and then actually putting it back for which I have not found a way to do that as Markdown parsing is kind of a one way road. Markdig has a markdown parse tree but it can't write out that tree once manipulated.

So long story short this depends on two things to which I currently have no answer

* A local grammar library for .NET (that can correlate to document position)
* A way to parse this back into the Markdown editor with highlighting (this can probably be figured out)
